src/cgutils.cpp

```diff
@@ -2017,7 +2102,7 @@ static Value *box_union(const jl_cgval_t &vinfo, jl_codectx_t *ctx, const SmallB
             // if this is a derived pointer, make sure the root usage itself is also visible to the delete-root pass
             mark_gc_use(vinfo);
         }
-        box_merge->addIncoming(emit_bitcast(vinfo.V, T_pjlvalue), defaultBB);
+        box_merge->addIncoming(builder.CreateLoad(vinfo.gcroot), defaultBB);
```

Comment:
Since you need to end up with a boxed value, I don't this they would ever disagree here.